bank/services/jpmorgan/jpmorgan.go

```diff
@@ -0,0 +1,232 @@
+	var response string
+	var respHeader map[string]interface{}
+	var status int
+	var err error
+	var bodyEncripted string
+
+	JPMorganURL := config.Get().URLJPMorgan
+	boleto.Title.BoletoType, boleto.Title.BoletoTypeCode = getBoletoType(boleto)
+
+	body := flow.NewFlow().From("message://?source=inline", boleto, templateRequest, tmpl.GetFuncMaps()).GetBody().(string)
+	head := hearders()
+
+	bodyEncripted, BoletoResponseError := b.encriptedBody(body)
+	if BoletoResponseError != nil {
+		return *BoletoResponseError, nil
+	}
+
+	b.log.Request(body, JPMorganURL, getLogRequestProperties(head, bodyEncripted))
+
+	duration := util.Duration(func() {
+		if config.Get().MockMode {
+			response, respHeader, status, err = util.PostWithHeader(JPMorganURL, body, config.Get().TimeoutDefault, head)
+		} else {
+			response, respHeader, status, err = util.PostTLSWithHeader(JPMorganURL, bodyEncripted, config.Get().TimeoutDefault, head, b.transport)
+		}
+
+	})
+	metrics.PushTimingMetric("jpmorgan-register-boleto-time", duration.Seconds())
+
+	b.log.Response(response, JPMorganURL, nil)
+
+	return mapJPMorganResponse(boleto, getContentType(respHeader), response, status, err), nil
+}
+
+func (b bankJPMorgan) encriptedBody(body string) (string, *models.BoletoResponse) {
+	errorResponse := models.GetBoletoResponseError("MP500", "Internal Error")
+	sk, err := certificate.GetCertificateFromStore(config.Get().AzureStorageJPMorganSignCrtName)
+	if err != nil {
+		return "", &errorResponse
+	}
+	bodyEncripted, err := b.jwtSigner.Sign(body, sk.([]byte))
+	if err != nil {
+		return "", &errorResponse
+	}
+	return bodyEncripted, nil
+}
+
+func getContentType(respHeader map[string]interface{}) string {
+	return respHeader["Content-Type"].(string)
+}
+
+func getLogRequestProperties(header map[string]string, encryptedBody string) map[string]string {
+	m := make(map[string]string)
+	m["encryptedBody"] = encryptedBody
+	for k, v := range header {
+		m[k] = v
+	}
+	return m
+}
+
+func hearders() map[string]string {
+	return map[string]string{"Content-Type": "text/xml"}
+}
+
+func getBoletoType(boleto *models.BoletoRequest) (bt string, btc string) {
+	return "DM", "02"
+}
+
+func mapJPMorganResponse(request *models.BoletoRequest, contentType string, response string, status int, httpErr error) models.BoletoResponse {
+	f := flow.NewFlow().To("set://?prop=body", response)
+	switch status {
+	case 200:
+		f.To("transform://?format=json", templateResponse, templateAPI, tmpl.GetFuncMaps())
+		f.To("unmarshall://?format=json", new(models.BoletoResponse))
+	case 0, 504:
+		return models.GetBoletoResponseError("MPTimeout", timeoutMessage(httpErr))
+	case 401, 500:
+		if "application/xml" == contentType {
+			f.To("set://?prop=body", parseXmlErrorToJson(response))
+			f.To("transform://?format=json", templateErrorXmltoJson, templateAPI, tmpl.GetFuncMaps())
+		} else if "application/json" == contentType {
```

Comment:
@davidsmedeiros Aqui ficou um _Yoda condition_
```suggestion
		if contentType == "application/xml"{
```